app/controllers/catalog_controller.rb

```diff
@@ -2166,39 +2180,74 @@ def automate_tree_needed?
     end
     playbook_details
   end
 
+  def fetch_terraform_template_details
+    terraform_template_details = {}
+    provision = @record.config_info[:provision]
+    provisioning_details = {}
+    provisioning_details[:repository] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptSource, provision[:repository_id])
+    provisioning_details[:template] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptPayload, provision[:configuration_script_payload_id])
+    provisioning_details[:credential] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential, provision[:credential_id])
+    provisioning_details[:network_credential] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::NetworkCredential, provision[:network_credential_id]) if provision[:network_credential_id]
+    provisioning_details[:execution_ttl] = provision[:execution_ttl]
+    provisioning_details[:verbosity] = provision[:verbosity]
+    provisioning_details[:log_output] = provision[:log_output]
+    provisioning_details[:become_enabled] = provision[:become_enabled] == true ? _('Yes') : _('No')
+    terraform_template_details[:provisioning] = provisioning_details
+    fetch_dialog(terraform_template_details, provision[:dialog_id], :provisioning)
+
+    # if @record.config_info[:retirement]
+    #   retirement = @record.config_info[:retirement]
+    #   terraform_template_details[:retirement] = {}
+    #   terraform_template_details[:retirement][:remove_resources] = retirement[:remove_resources]
+    #   if retirement[:repository_id]
+    #     terraform_template_details[:retirement][:repository] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptSource, retirement[:repository_id])
+    #     terraform_template_details[:retirement][:template] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptPayload, retirement[:configuration_script_payload_id])
+    #     terraform_template_details[:retirement][:credential] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential, retirement[:credential_id])
+    #     terraform_template_details[:retirement][:network_credential] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::NetworkCredential, retirement[:network_credential_id]) if retirement[:network_credential_id]
+    #   end
+    #   terraform_template_details[:retirement][:execution_ttl] = retirement[:execution_ttl]
+    #   terraform_template_details[:retirement][:verbosity] = retirement[:verbosity]
+    #   terraform_template_details[:retirement][:log_output] = retirement[:log_output]
+    #   terraform_template_details[:retirement][:become_enabled] = retirement[:become_enabled] == true ? _('Yes') : _('No')
+    # end
```

Comment:
Minor but I think we should add a comment here about why this block of code is commented out